service/raft/raft_group0.cc

```diff
@@ -894,7 +894,7 @@ future<group0_peer_exchange> raft_group0::peer_exchange(discovery::peer_list pee
             co_return group0_peer_exchange{group0_info{
                 .group0_id = std::get<raft::group_id>(_group0),
                 .id = _raft_gr.group0().id(),
-                .ip_addr = _raft_gr.address_map().get_inet_address(_raft_gr.group0().id())
+                .ip_addr = _raft_gr.address_map().find(_raft_gr.group0().id()).value_or(gms::inet_address{}),
```

Comment:
Continuing our discussion from earlier https://github.com/scylladb/scylladb/pull/11635/commits/bf635b9d7e063d0b5c3aa32539b2b38a6dd643d5#r1019092762

> So do you suggest using get_broadcast_address instead?

Yes, actually, that would make my question irrelevant. Whether we have our own IP in the address map or not, we will always be able to return our own IP.

So I agree we should return `.ip_addr = get_broadcast_address()`.

Reply:
Alternatively - is this `ip_addr` field even necessary?

Whoever sent the `peer_exchange` request - they obviously know the destination IP they sent it to. They can just reuse it. We're returning redundant information, no?

I guess it doesn't hurt to return `ip_addr` here, maybe it makes the code shorter on the sender side.

Reply:
It makes life easier for the receiver. I'll use get_broadcast_addr()

Reply:
So the history of this is that it would initially return the leader id and address. This was removed when forwarding was added, assuming our own address is just as good as hte leader's. this adds an extra hop. But having the ip and addr explicitly in response also adds hope that the hop can be removed in the future.

Reply:
Used _gossiper.get_broadcast_address()